lib/OpenQA/Schema/Result/Jobs.pm

```diff
@@ -499,21 +502,29 @@ sub to_hash {
     if ($job->assigned_worker_id) {
         $j->{assigned_worker_id} = $job->assigned_worker_id;
     }
-    if (my $origin = $job->origin) {
+    if (exists $args{origin}) {
+        $j->{origin_id} = $args{origin} if $args{origin};
+    }
+    elsif (my $origin = $job->origin) {
         $j->{origin_id} = $origin->id;
     }
     if (my $reason = $job->reason) {
         $j->{reason} = $reason;
     }
-    $j->{settings} = $job->settings_hash;
+    $j->{settings} = $job->settings_hash($settings);
     # hashes are left for script compatibility with schema version 38
     $j->{test} = $job->TEST;
     if ($args{assets}) {
         my $assets = $job->{_assets} // [map { $_->asset } $job->jobs_assets->all];
         push @{$j->{assets}->{$_->type}}, $_->name for @$assets;
     }
     if ($args{deps}) {
-        $j = {%$j, %{$job->dependencies}};
+        if ($dependencies) {
+            $j = {%$j, %{$job->dependencies($dependencies->{children} || [], $dependencies->{parents} || [])}};
+        }
+        else {
+            $j = {%$j, %{$job->dependencies()}};
+        }
```

Comment:
```suggestion
        $j = {%$j, $dependencies ? %{$job->dependencies($dependencies->{children} || [], $dependencies->{parents} || [])} : %{$job->dependencies()}};
```

but the amount of curly braces is hardly readable in both cases. Maybe better extract a function to return what is needed here

Reply:
I rearranged it a bit